### PR TITLE
Add sapinstance-hostname-resolver gatherer doc

### PR DESCRIPTION
**Pull request by rtorrero**
Add sapinstance_hostname_resolver gatherer documentation, added in [PR#274](https://github.com/trento-project/agent/pull/274)

### CODE REVIEW

File: guides/gatherers.md [hunk 1]
```diff
@@ -791,6 +790,56 @@ Example output (in Rhai):
 ];
 ```
 
+### sapinstance_hostname_resolver
+
+**Argument required**: no.
+
+This gatherer uses the filesystem to search for sap systems using the discovered profile file names to get the virtual hostnames associated to each
```
Comment (arbulu89) [inline guides/gatherers.md:797]: `sap` -> `SAP`

File: guides/gatherers.md [hunk 2]
```diff
@@ -791,6 +790,56 @@ Example output (in Rhai):
...
+instance of the sap system. It will then attempt to resolve those hostnames to confirm that they are resolvable and afterwards it will attempt a ping
```
Comment (arbulu89) [inline guides/gatherers.md:798]: Let's try to avoid future term. `It then attempts to resolve...`

File: guides/gatherers.md [hunk 3]
```diff
@@ -791,6 +790,56 @@ Example output (in Rhai):
...
+    #{
+      "addresses": [
+        "1.1.1.82"
+      ],
+      "hostname": "sapqasas",
+      "instance_name": "ASCS00",
+      "reachability": true
+    },
+    #{
+      "addresses": (),
```
Comment (arbulu89) [inline guides/gatherers.md:824]: Maybe we can put an example on top like
```
// hostname not resolvable
```

**Review by arbulu89 — APPROVED** [review]
Looks good!

**Review by EMaksy — APPROVED** [review]
Hey LGTM, besides @arbulu89  comments nothing to add :)

**Pull request merged**